### ISSUE TITLE
build: Need options to specify BDB location

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#3921]
Many people compile BerkeleyDb 4.8 themselves and understandably don't want to install it system-wide.

I'd suggest the following names:

```
--with-bdb-libdir=
--with-bdb-includedir=
```

One can always override LDFLAGS and CFLAGS to find them manually, don't forget that you need to manualy specify optimization flags in that case ie: 

```
./configure LDFLAGS="-L/path/to/libs" CPPFLAGS="-I/path/to/includes -O2"
```

But it'd be easier if there was a flag for this as it is very common now.

Alternative would be `BDB_CFLAGS` and `BDB_LIBS`, analogous to `SSL_CFLAGS` and `SSL_LIBS`.

**Comment by laanwj** [comment]
I'm trying to build Bitcoin Core on FreeBSD and would be very much helped by this. Also in our build guides we routinely override `LDFLAGS`/`CXXFLAGS` for this purpose, which shouldn't be necessary if there was an option or a specific env variable for this:

- https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#berkeley-db
- https://github.com/bitcoin/bitcoin/blob/master/doc/build-openbsd.md#building-bitcoin-core

Anyone up for this? I can try to do it myself but I don't know that much about m4 and will probably come up with a hacky solution.

**Comment by laanwj** [comment]
Fixed by #9705
